### PR TITLE
[13.0][MIG] purchase_operating_unit

### PR DESCRIPTION
**Pull request by BT-nstuder**
## Proposed changes

I've migrated the module `purchase_operating_unit` and it's functionalities from Odoo `12.0` to `13.0`. I've checked if the all unit tests are still running.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Migration Update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst) doc
- [x] I have signed the [CLA](https://odoo-community.org/page/cla)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated necessary documentation

## Further comments

Because this module interferes with the accounting, I had to change the things proposed in this GitHub [Issue](https://github.com/OCA/maintainer-tools/issues/430). Some of the methods of Odoo `12.0` got removed, so I rewrote some of the methods.

* `_onchange_operating_unit_id` in `account_invoice.py`
* `_onchange_purchase_auto_complete` in `account_invoice.py`

The unit tests had to be adapted as well. Because the `purchase_id` of an `account.move` is not stored anymore, I used the actual `ID` of the `accound.move` to check if test users have access to it or not. 

This module depends on the `stock_operating_unit` module. That means that there should be a migration of it to `13.0`. Because there is already a PR for a migration I've added the changes of the PR to my local environment to test my migration.

This PR depends on the current migration PR #231.

### CODE REVIEW

**Review by alan196 — CHANGES_REQUESTED** [review]
#231 is merged and you have conflicts with .sort.cfg

@BT-nstuder  Please rebase this branch

**Comment by BT-nstuder** [comment]
Hello @alan196, the branch is rebased.

File: purchase_operating_unit/models/account_invoice.py [hunk 1]
```diff
@@ -5,35 +5,37 @@
...
-        if self.purchase_id and self.purchase_id.operating_unit_id:
+        purchase_id = self.purchase_id
+        if self.purchase_vendor_bill_id.purchase_order_id:
+            purchase_id = self.purchase_vendor_bill_id.purchase_order_id
+        if purchase_id and purchase_id.operating_unit_id:
             # Assign OU from PO to Invoice
-            self.operating_unit_id = self.purchase_id.operating_unit_id.id
-        return super(AccountInvoice, self).purchase_order_change()
+            self.operating_unit_id = purchase_id.operating_unit_id.id
+        return super(AccountMove, self)._onchange_purchase_auto_complete()
```
Comment (alan196) [inline purchase_operating_unit/models/account_invoice.py]:
```suggestion
        return super()._onchange_purchase_auto_complete()
```

**Review by alan196 — CHANGES_REQUESTED** [review]

**Comment by alan196** [comment]
Tested functionally 👍 just a little change to approve

**Review by AaronHForgeFlow — APPROVED** [review]
Functional speaking works as expected :+1:

**Review by alan196 — APPROVED** [review]
👍 LGTM

**Pull request merged**